// @skill-wallet/auth/dist/collection/components/skillwallet-auth/user-role.js
import { Component, h, State, Prop } from '@stencil/core';
export class UserRole {
  constructor() {
    this.roleSelected = null;
    this.isLoading = false;
    this.buttonClass = 'disabled';
    this.skill = 10;
    this.communityAddress = null;
    this.partnersAddress = null;
  }
  render() {
    return (h("div", { class: "user-role-modal-window-child" }, this.isPartner ?
      h("roles-screen-partner", { community: this.community, isPartner: this.isPartner, isLoading: this.isLoading, buttonClass: this.buttonClass, roleSelected: this.roleSelected, communityAddress: this.communityAddress, partnersAddress: this.partnersAddress }) :
      h("roles-screen-new-user", { skill: this.skill, community: this.community, isPartner: this.isPartner, isLoading: this.isLoading, buttonClass: this.buttonClass, roleSelected: this.roleSelected })));
  }
  static get is() { return "user-role"; }
  static get encapsulation() { return "shadow"; }
  static get originalStyleUrls() { return {
    "$": ["skillwallet-auth.css"]
  }; }
  static get styleUrls() { return {
    "$": ["skillwallet-auth.css"]
  }; }
  static get properties() { return {
    "community": {
      "type": "any",
      "mutable": false,
      "complexType": {
        "original": "any",
        "resolved": "any",
        "references": {}
      },
      "required": false,
      "optional": false,
      "docs": {
        "tags": [],
        "text": ""
      },
      "attribute": "community",
      "reflect": false
    },
    "isPartner": {
      "type": "unknown",
      "mutable": false,
      "complexType": {
        "original": "Boolean",
        "resolved": "Boolean",
        "references": {
          "Boolean": {
            "location": "global"
          }
        }
      },
      "required": false,
      "optional": false,
      "docs": {
        "tags": [],
        "text": ""
      }
    },
    "communityAddress": {
      "type": "string",
      "mutable": false,
      "complexType": {
        "original": "string",
        "resolved": "string",
        "references": {}
      },
      "required": false,
      "optional": false,
      "docs": {
        "tags": [],
        "text": ""
      },
      "attribute": "community-address",
      "reflect": false,
      "defaultValue": "null"
    },
    "partnersAddress": {
      "type": "string",
      "mutable": false,
      "complexType": {
        "original": "string",
        "resolved": "string",
        "references": {}
      },
      "required": false,
      "optional": false,
      "docs": {
        "tags": [],
        "text": ""
      },
      "attribute": "partners-address",
      "reflect": false,
      "defaultValue": "null"
    }
  }; }
  static get states() { return {
    "roleSelected": {},
    "isLoading": {},
    "buttonClass": {},
    "skill": {}
  }; }
}
