// @skill-wallet/auth/dist/collection/validators/validator.factory.js
import { FileValidator, getLengthValidator } from './validator';
// Thanks to @Sébastien.Lubineau https://medium.com/@sebastien.lubineau/input-validation-with-stenciljs-f464644545c8
export var ValidatorsName;
(function (ValidatorsName) {
  ValidatorsName["file"] = "file";
  ValidatorsName["length"] = "length";
})(ValidatorsName || (ValidatorsName = {}));
export const defaultValidator = {
  validate: (_x) => true
};
export function getValidator(v) {
  if (typeof v === 'string') {
    return validatorFactory(v, null);
  }
  else if (typeof v === 'object' && (v.name === 'length' || v.name === 'file')) {
    return validatorFactory(v.name, v.options);
  }
}
// simple controller for handling our typed validators
export function validatorFactory(name, options) {
  options ? options : {};
  switch (name) {
    case (ValidatorsName.file):
      return FileValidator(options);
    case (ValidatorsName.length):
      return getLengthValidator(options.min, options.max);
    default:
      return defaultValidator;
  }
}
